scripts/o3de/o3de/project_manager_interface.py

```diff
@@ -0,0 +1,300 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+Contains functions for the project manager to call that gather data from o3de scripts and massage it
+"""
+
+import logging
+import pathlib
+
+from o3de import cmake, disable_gem, download, enable_gem, engine_properties, engine_template, manifest, project_properties, register, repo
+
+logger = logging.getLogger('o3de.project_manager_interface')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+
+def get_engine_info(engine_name: str = None, engine_path: str or pathlib.Path = None) -> dict or None:
+    """
+        If engine_path is provided then continue
+        Elif only engine_name is provided then call get_registered and use returned path as engine_path
+        Elif neither parameter is provided call get_this_engine_path and use returned path as engine_path
+
+        Call get_engine_json_data using engine_path to populate initial engine data
+
+        Call load_o3de_manifest or get default folders for gems, projects, ect. using get_o3de_..._folder
+        Insert either default folder from manifest or get_o3de_..._folder into engine dict for all folders
+
+        Call get_manifest_engines and check that engine is in the list if it is set registered attr to True
+
+        :param engine_name: Engine name to find engine path for and get json data for
+        :param engine_path: Engine path to gather json data for
+
+        :return dict containing engine info if it was found. Otherwise None is returned
+    """
+    return dict()
+
+
+def set_engine_info(engine_info: dict) -> int:
```

Comment:
We have an opportunity to standardize how we handle errors.
Currently it looks like our "getter" functions return an object type and throw an exception on error, while our "setter" functions return an integer exit code.   If these exit codes are meaningful then I'm cool with keeping them, but if the only use is a value of 0 for success and all other values are failures then we might as well just use exceptions for failures.

Reply:
Removed all int exit code returns, we will instead throw exceptions if a non zero exit code is returned by an internal python function and listen for them in cpp.